modules/rest/src/main/java/com/enonic/xp/app/contentstudio/service/ContentVisitorServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.enonic.xp.app.contentstudio.service;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+import org.osgi.service.component.annotations.Component;
+
+import com.enonic.xp.app.contentstudio.json.content.ContentVisitorParams;
+import com.enonic.xp.app.contentstudio.rest.resource.auth.json.UserJson;
+import com.enonic.xp.content.ContentId;
+import com.enonic.xp.context.ContextAccessor;
+import com.enonic.xp.security.User;
+
+@Component(service = ContentVisitorService.class)
+public class ContentVisitorServiceImpl
```

Comment:
VisitorServiceImpl has sufficient information about visitors. This is double bookkeeping that may create inconsistency.